### ISSUE TITLE
Use your trained model to predict the pictures in the example, but there is no prediction result

### ISSUE THREAD

**Issue by miracle1111111** [Kohulan/DECIMER-Image-to-SMILES#5]
I followed the following steps：
$ pip install tensorflow pillow deepsmiles
$ git clone https://github.com/Kohulan/DECIMER-Image-to-SMILES decimer
$ cd decimer/Network
$ python Predictor.py --input ../Sample_Images/P_59687969.png
![image](https://user-images.githubusercontent.com/49110370/100881143-8b5b3700-34e8-11eb-9f3f-ca197d666695.png)
![image](https://user-images.githubusercontent.com/49110370/100881464-edb43780-34e8-11eb-901f-09fcf5687058.png)

In the end, Predictions.txt has no prediction results. I don’t know what went wrong, I am very eager to know why, thank you

**Comment by Kohulan** [comment]
@miracle1111111 Thank you for the report. The code has been updated. Please download the model from the link given.

**Comment by miracle1111111** [comment]
Hello . I used the downloaded model to predict, and the following error was prompted

![image](https://user-images.githubusercontent.com/49110370/101023114-44834500-35ad-11eb-8dc3-f07569ce87c9.png)
I watched it for a long time, but I didn't find out where the problem was. I hope the author can check Related code so that it can successfully predict a single picture example.
In addition, in I2S_evalData.py, line 62 I2S_Data.data_loader(PATH,total_data) has 7 return values. Should I correct this 
![image](https://user-images.githubusercontent.com/49110370/101024163-aee8b500-35ae-11eb-8f18-589154a9ad16.png)

Thank you for your answer.

**Comment by Kohulan** [comment]
@miracle1111111 I have updated the instructions. Try to clone the code again and follow the instructions under the Network folder. 

I2S_evaldata script corrected, but it has nothing to do with the error produced here.

**Comment by miracle1111111** [comment]
Thank you very much, it can be predicted. In addition, can your final DeepSMILES be made public? I want to know if there are spaces in DeepSMILES, because the generated dictionary is problematic if it does not.
191914,CCSC=CC=CC=C6)))N
or
191914,C C S C = C C = C C = C 6 ) ) ) N

**Comment by Kohulan** [comment]
@miracle1111111 The Predictions contains spaces, I am removing the spaces at the end of predictions before I parse it through the DeepSMILES decoder. Hope it helps!

**Comment by miracle1111111** [comment]
Thank you very much for your answer, it is very helpful to me. I am going to run with my own data set, if I encounter problems, I hope I can ask you for advice.

**Comment by miracle1111111** [comment]
Hello, I used my own data set to train the model. After changing the model and the path of tokenizer.pickle, Predictor.py prompts the following error:


I don’t know if it’s in the Trainer_Image2Smiles.py file. You don’t have the code to generate the tokenizer.pkl file. I added it in Trainer_Image2Smiles.py&nbsp;&nbsp;with&nbsp;open('/home/xyji/decimer1/tokenizer.pickle',&nbsp;'wb')&nbsp;as&nbsp;handle:&nbsp;
&nbsp;&nbsp;&nbsp;&nbsp;pickle.dump(tokenizer,&nbsp;handle,&nbsp;protocol=pickle.HIGHEST_PROTOCOL)&nbsp;
I like your work very much. I have encountered some difficulties. I hope I can get your help.



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Kohulan/DECIMER-Image-to-SMILES"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年12月3日(星期四) 晚上10:40
收件人:&nbsp;"Kohulan/DECIMER-Image-to-SMILES"<DECIMER-Image-to-SMILES@noreply.github.com&gt;;
抄送:&nbsp;"Miracle"<541766011@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [Kohulan/DECIMER-Image-to-SMILES] Use your trained model to predict the pictures in the example, but there is no prediction result (#5)





 
Closed #5.

**Comment by miracle1111111** [comment]
Can you publish your SMILES or DEEPSMILES data set, because I see that your paper has many restrictions on molecules. Thank you

**Comment by Kohulan** [comment]
@miracle1111111 Thank you for your interest :), The Paper is a "Preliminary communication", we will publish the data with our main paper.

**Comment by miracle1111111** [comment]
Thank you. In addition, the code you provided does not use the validation set to adjust the parameters? I found that Evaluate.py and Predictor.py are both predictions, and both use the training set to train the model to predict.

**Comment by Kohulan** [comment]
@miracle1111111 Yes, We didn't use a validation set, The parameters mentioned in the code are the values we used after optimizing the network. Evaluate.py used for evaluating the results in batches. Predictor.py is for testing the model easily from the command line. The model was tested using a test set of unseen data.

**Comment by miracle1111111** [comment]
Thinks, finally I want to know whether your method is more suitable for shorter molecules or a large amount of data, because when I use about 80,000 data, the length of DeepSMILES is between 10-100, the trained model  performance is poor.

**Comment by Kohulan** [comment]
You should use images of molecules with the maximum SMILES character length of 40, We are currently working on the improvements.

**Comment by miracle1111111** [comment]
I understand, thank you for your reply.

**Comment by miracle1111111** [comment]
Hello, the Valid SMILES in the paper refers to the predicted DeepSMILES can be correctly parsed as the proportion of SMILES？In addition, the paper said that training to 25 epochs usually makes the model converge, but I train for about 600 rounds before the model gradually converges.thinks

![image](https://user-images.githubusercontent.com/49110370/102316221-800c0f00-3fb0-11eb-8d53-c335d0df933c.png)
